packages/neuron-wallet/tests/utils/export-history/index.test.ts

```diff
@@ -1,29 +1,147 @@
+  }
+
+  beforeAll(() => {
     i18n.changeLanguage('en')
-    await initConnection(":memory:")
-    for (const fixture of Object.values(fixtures.transactions)) {
-      const tx = Transaction.fromSDK(
-        fixture.transaction as any,
-        fixture.transaction as any,
-      )
-      tx.outputs.forEach((o, i) => o.daoData = (fixture.transaction.outputs[i] as any).daoData)
-      await TransactionPersistor.saveFetchTx(tx)
+  })
+
+  beforeEach(async () => {
+    await initConnection()
+    return saveTransactions(transactions)
+  })
+
+  afterEach(() => {
+    try {
+      fs.unlinkSync(stubProvider.filePath)
+    } catch {
+      // ignore
     }
+    stubProvider.walletID = WALLET_ID
+    stubProvider.chainType = MAINNET_CHAIN_TYPE
+    stubProvider.filePath = FILE_PATH
+    return closeConnection()
   })
 
-  afterAll(() => {
-    return getConnection().close()
+  describe('When wallet id is missing', () => {
+    beforeEach(() => {
+      stubProvider.walletID = ''
+    })
+
+    it("Should throw an error", async () => {
+      expect.assertions(1)
+      try {
+        await exportHistory(stubProvider)
+      } catch (err) {
+        expect(err).toEqual(new Error('Wallet ID is required'))
+      }
+    })
+  })
+
+  describe('When filePath is missing', () => {
+    beforeEach(() => {
+      stubProvider.filePath = ''
+    })
+
+    it('Should throw an error', async () => {
+      expect.assertions(1)
+      try {
+        await exportHistory(stubProvider)
+      } catch (err) {
+        expect(err).toEqual(new Error('File Path is required'))
+      }
+    })
   })
 
-  it.skip('Test export to csv', async () => {
+  describe('When file exists', () => {
+    const originalExistsSync = fs.existsSync
+    const originalUnlinkSync = fs.unlinkSync
+
+    afterAll(() => {
+      fs.existsSync = originalExistsSync
+      fs.unlinkSync = originalUnlinkSync
+    })
+
+    beforeEach(() => {
+      fs.existsSync = jest.fn(originalExistsSync).mockReturnValueOnce(true)
+      fs.unlinkSync = jest.fn(originalUnlinkSync).mockReturnValue()
+    })
+
+    it('Should remove file first', async () => {
+      expect.assertions(2)
+      await exportHistory(stubProvider)
+      expect(fs.existsSync).toHaveBeenCalledWith(stubProvider.filePath)
+      expect(fs.unlinkSync).toHaveBeenCalledWith(stubProvider.filePath)
+    })
+  })
+
+  describe('When no transactions exported', () => {
+    describe('When it\'s Mainnet', () => {
+      beforeEach(() => {
+        stubProvider.chainType = MAINNET_CHAIN_TYPE
+      })
+
+      it('Should export table header without sudt column', async () => {
+        expect.assertions(2)
+        const totalCount = await exportHistory(stubProvider)
+        const actual = fs.readFileSync(stubProvider.filePath, 'utf8')
+        expect(totalCount).toBe(0)
+        expect(actual).toBe('Time,Block Number,Transaction Hash,Transaction Type,CKB Amount,Description\n')
+      })
+    })
+
+    describe('When it\'s Mainnet', () => {
```

Comment:
I guess the wording here is `when it's testnet`

Reply:
Updated in https://github.com/nervosnetwork/neuron/pull/1738/commits/a4d8f07350da45d47b237333eeab458e26822346